trin-core/src/portalnet/types.rs

```diff
@@ -220,113 +219,61 @@ impl Request {
 pub enum Response {
     Pong(Pong),
     Nodes(Nodes),
-    FoundContent(FoundContent),
+    Content(Content),
 }
 
 impl Response {
     fn message_id(&self) -> u8 {
         match self {
             Response::Pong(_) => 2,
             Response::Nodes(_) => 4,
-            Response::FoundContent(_) => 6,
+            Response::Content(_) => 6,
         }
     }
 }
 
 /// Custom payload element of Ping and Pong messages
 #[derive(Debug, PartialEq, Clone, Encode, Decode)]
-pub struct CustomPayload {
-    /// Overlay data radius
-    pub data_radius: U256,
-    /// Optional payload element of SSZ type List[uint8, max_length=2048].
-    pub payload: Option<ByteList>,
-}
+pub struct CustomPayload;
 
 impl CustomPayload {
-    pub fn new(data_radius: U256, payload: Option<Vec<u8>>) -> Self {
-        match payload {
-            Some(payload) => {
-                let ssz_list = VariableList::from(payload);
-                let message: ByteList = ByteList::from(ssz_list);
-
-                Self {
-                    data_radius,
-                    payload: Some(message),
-                }
-            }
-            None => Self {
-                data_radius,
-                payload: None,
-            },
-        }
+    pub fn new(payload: Vec<u8>) -> ByteList {
```

Comment:
nit: Not sure why we need the `CustomPayload` type if it is a unit struct. Perhaps instead we can implement `From<Vec<u8>>` for `ByteList`?

Reply:
Nice, it was a hangover from the previous hacky union implementation, but turns out it's already implemented